ADR009: Non-Interactive Default Rules for Reduced Padding (#1003)
celestiaorg/celestia-app: docs/architecture/adr-008-square-size-independent-message-commitments.md

### Negative

1. The amount of subtree roots per commitment is O(sqrt(n)), while n is the number of message shares. The worst case for the number of subtree roots is depicted in the diagram below - an entire block missing one share.

>   ![Interactive Commitment 2](./assets/complexity.png)  

Author: 2 spaces are the same as an enter in markdown. If we do not have that then the text will continue after the picture and not in a new line. It looks nicer with the 2 spaces.

>   ![Interactive Commitment 2](./assets/complexity.png)  

Reviewer: Oh, then
```suggestion

![Interactive Commitment 2](./assets/complexity.png)  

```

>   ![Interactive Commitment 2](./assets/complexity.png)  

Author: That does not work because then the enumeration breaks in which the picture is in. Line breaks inside enumerations is a heated topic. https://meta.stackexchange.com/a/192898 I did not get Option 1 (from stack overflow) to work, because of bad indentation and makrdownlint complaining.

  - Currently, prepare proposal performs [`estimateSquareSize`](https://github.com/rootulp/celestia-app/blob/6f3b3ae437b2a70d72ff6be2741abb8b5378caa0/app/estimate_square_size.go#L98-L101) prior to splitting PFBs into shares because the square size is needed to malleate PFBs and extract the appropriate message share commitment for a particular square size. Since malleation no longer requires a square size, it may be possible to remove square size estimation which renders the following issues obsolete:
    - <https://github.com/informalsystems/audit-celestia/issues/12>
    - <https://github.com/informalsystems/audit-celestia/issues/24>

> - Inter-message padding can be reduced in the worst case by 50% if we can change the non-interactive default rules. An in-depth analysis is performed at [ADR 009: New Non-Interactive Default Rules for Reduced Padding](Link).

Reviewer 2: It looks like this link is causing markdownt lint link checking to fail.

```suggestion
- Inter-message padding can be reduced in the worst case by 50% if we can change the non-interactive default rules. An in-depth analysis is performed at [ADR 009: New Non-Interactive Default Rules for Reduced Padding](./adr-009-proof-size-analysis.md).
```

> - Inter-message padding can be reduced in the worst case by 50% if we can change the non-interactive default rules. An in-depth analysis is performed at [ADR 009: New Non-Interactive Default Rules for Reduced Padding](./adr-009-proof-size-analysis.md).

Reviewer 2: dang, sorry but markdownlint now complains because `./adr-009-proof-size-analysis.md` doesn't exist

https://github.com/celestiaorg/celestia-app/actions/runs/3486753915/jobs/5833589081#step:4:223

I wonder if this will fix:

```suggestion
- Inter-message padding can be reduced in the worst case by 50% if we can change the non-interactive default rules. An in-depth analysis is performed at [ADR 009: New Non-Interactive Default Rules for Reduced Padding](./adr-009-non-interactive-default-rules-for-reduced-padding.md).
```
